Dz10

angryQ opened the pull request.

angryQ committed:
initial

angryQ committed:
Набросок CollectionView

Нужно разобраться почему в ячейке отображаются не все элементы

angryQ committed:
Теперь отображабтся все лэйблы в ячейке

 Но думаю я все поменяю. Что-то мне не нравится такая реализация.

angryQ committed:
Добавила все задания

Снова проблемы с нотником, работала когда была возможность

angryQ committed:
Удалила лишнее

(Пробовала создать лэйаут скролл вью с помощью таблицы или полностью кодом)

ios-review commented on DZ10/DZ10/Collection/CollectionViewController.swift:
```diff
@@ -0,0 +1,75 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.modalPresentationStyle = .fullScreen
```
данное свойство необходимо устанавливать за пределами контроллера до вызова `presentingViewController.present(thisViewController)`

ios-review commented on DZ10/DZ10/Collection/imageCollection.swift:
```diff
@@ -0,0 +1,31 @@
+class ImageCollection {
+
+    let images = [
+        Item(name: "Картинка #1", image: UIImage(named: "pic-10")!, oldPrice: 13500, newPrice: 4900, discount: "65%"),
```
В данном случае discount надежнее считать.
Почему?

ios-review commented on DZ10/DZ10/Grid2x2CodeViewController.swift:
```diff
@@ -0,0 +1,72 @@
+
+import UIKit
+
+class Grid2x2CodeViewController: UIViewController {
```
Эти задания необходимо было выполнить в Interface Builder (storyboard, xib).
Используя autolayout - констрейнты.
В данном случае без UIStackView.

ios-review commented on DZ10/DZ10/Grid2x2CodeViewController.swift:
```diff
@@ -0,0 +1,72 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
```
Пустой метод не нужен.

ios-review commented on DZ10/DZ10/Grid2x2CodeViewController.swift:
```diff
@@ -0,0 +1,72 @@
+        let viewHeightWithoutNavigationBar = self.view.safeAreaLayoutGuide.layoutFrame.size.height - navigationBarHeight
+        let paddingLeftForHolderView: CGFloat = 10
+        
+        self.view.addSubview(holderView)
```
Вью нельзя добавлять в момент вычисления layout.
layout по сути вычисляет расположение и размер всех вьюх.
Если в этот момент добавить новую, то придется снова считать. И так по кругу до бесконечности.

Этот код может быть вызван несколько раз: экран загрузился - вызвался, потом пользователь повернул экран - снова вызвался.

ios-review commented on DZ10/DZ10/Grid3x3CodeViewController.swift:
```diff
@@ -0,0 +1,73 @@
+
+import UIKit
+
+class Grid3x3CodeViewController: UIViewController {
```
Реализовать в сториборде на констрейнтах.

ios-review commented on DZ10/DZ10/GridCustomCodeViewController.swift:
```diff
@@ -0,0 +1,66 @@
+        createGrid()
+    }
+    
+    private func createGrid() {
```
over engineering ))

ios-review commented on DZ10/DZ10/LayoutTaskVideoViewController.swift:
```diff
@@ -0,0 +1,59 @@
+
+import UIKit
+
+class LayoutTaskVideoViewController: UIViewController {
```
👌👍🔥

ios-review commented on DZ10/DZ10/Table/MyTableViewController.swift:
```diff
@@ -0,0 +1,122 @@
+
+import UIKit
+
+struct Setting {
```
👍

ios-review commented on DZ10/DZ10/Table/MyTableViewController.swift:
```diff
@@ -0,0 +1,122 @@
+        }
+    }
+    //Почему-то не работает
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        if section == 0 {
+            return 5
+        }
+        else {
+            return tableView.rowHeight
+        }
+    }
```
Это метод делегата.
Делегат для таблицы не установлен.
https://developer.apple.com/documentation/uikit/uitableviewdelegate/1614855-tableview

ios-review replied:
Поэтому не срабатывал)

ios-review commented on DZ10/DZ10/Table/MyTableViewController.swift:
```diff
@@ -0,0 +1,122 @@
+    }
+    
+    //метод не мой, взяла с интернета, меняю им размер картинки
+    func imageWithImage(image: UIImage, scaledToSize newSize: CGSize) -> UIImage {
+        
+        UIGraphicsBeginImageContext(newSize)
+        image.draw(in: CGRect(x: 0, y: 0, width: newSize.width, height: newSize.height))
+        let newImage = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        return newImage!.withRenderingMode(.alwaysTemplate)
+    }
+}
+extension UIColor {
+    static var random: UIColor {
+        return UIColor(red: .random(in: 0...1),
+                       green: .random(in: 0...1),
+                       blue: .random(in: 0...1),
+                       alpha: 1.0)
+    }
```
🔥

ios-review commented on DZ10/DZ10/Table/MyTableViewController.swift:
```diff
@@ -0,0 +1,122 @@
+        cell.commentLabel.text = setting.comment
+        cell.accessoryType = setting.accessoryType
+        
+        if indexPath.section == 2 && indexPath.row == 0 {
+            cell.commentLabel.backgroundColor = .red
+            cell.commentLabel.layer.cornerRadius = 10
+            cell.commentLabel.layer.masksToBounds = true
+            cell.commentLabel.textColor = .white
+            cell.mySwitch.isHidden = true
+            cell.accessoryType = .disclosureIndicator
+        }
```
В cellForRow не должно быть проверок на section и row.
Все можно вынести в модель данных.

ios-review commented on DZ10/DZ10/Table/SettingTableViewCell.swift:
```diff
@@ -0,0 +1,35 @@
+        }
+    }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        // Configure the view for the selected state
+    }
```
пустые методы

ios-review commented on DZ10/DZ10/ViewController.swift:
```diff
@@ -0,0 +1,17 @@
+//
+//  ViewController.swift
+//  DZ10
+//
+//  Created by Kristina Kudinova on 17.11.2020.
+//  Copyright © 2020 Kristina Kudinova. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+}
+
```
пустой файл

ios-review reviewed (changes requested):
Интересная реализация дополнительного задания LayoutTaskVideoViewController.
Каждый раз удивляюсь возможным решениям.
Молодец👍

В комментарии доработки.

ios-review commented:
Еще такое:
<img width="1551" alt="Screenshot 2020-12-23 at 20 41 10" src="https://user-images.githubusercontent.com/62285448/103003314-22a13080-4563-11eb-81a5-614965d7e7be.png">

angryQ committed:
Переделала доп.задания за прошлый модуль

До этого лэйаут создавался полностью кодом, теперь все в сториборде констейнтрами

angryQ committed:
Вынесла проверку секции таблицы из cellForRow в структуру Settings

angryQ committed:
Таблица. Решила перенести стиль лэйбла  в енам

ios-review commented on DZ10/DZ10/Table/MyTableViewController.swift:
```diff
@@ -64,20 +65,21 @@ extension MyTableViewController: UITableViewDataSource {
+        cell.commentLabel.backgroundColor = setting.commentLabelStyle?.rawValue.backgroundColor
+        if let corderRadius = setting.commentLabelStyle?.rawValue.cornerRadius {
+            cell.commentLabel.cornerRadius = corderRadius
         }
```
Чуть короче:
`cell.commentLabel.cornerRadius = setting.commentLabelStyle?.rawValue.cornerRadius ?? 0`

ios-review commented on DZ10/DZ10/Table/MyTableViewController.swift:
```diff
@@ -64,20 +65,21 @@ extension MyTableViewController: UITableViewDataSource {
+        if let corderRadius = setting.commentLabelStyle?.rawValue.cornerRadius {
+            cell.commentLabel.cornerRadius = corderRadius
         }
+        if let maskToBounds = setting.commentLabelStyle?.rawValue.maskToBounds {
+            cell.commentLabel.layer.masksToBounds = maskToBounds
+        }
```
`cell.commentLabel.layer.masksToBounds = setting.commentLabelStyle?.rawValue.maskToBounds ?? false`

ios-review commented on DZ10/DZ10/Table/MyTableViewController.swift:
```diff
@@ -86,13 +88,9 @@ extension MyTableViewController: UITableViewDataSource {
-        } else {
-            return " "
-        }
+        return " "
```
Это можно не реализовывать вовсе.

ios-review replied:
```
func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         if section == 0 {
             return 5
        }
        else {
            return tableView.rowHeight
        }
    }
```

Вот здесь проверьте, чему равен tableView.rowHeight?
И попробуйте установить целое положительное значение)

ios-review commented on DZ10/DZ10/Table/MyTableViewController.swift:
```diff
@@ -120,3 +119,15 @@ extension UIColor {
     }
 }
+
+enum CommentLabelType {
+    case standard
+    case red
+    
+    var rawValue: (textColor: UIColor, backgroundColor: UIColor, cornerRadius: CGFloat, maskToBounds: Bool) {
+        switch self {
+            case .standard: return (textColor: UIColor.gray, backgroundColor: UIColor.clear, cornerRadius: 0, maskToBounds: false)
+            case .red: return (textColor: UIColor.white, backgroundColor: UIColor.red, cornerRadius: 10, maskToBounds: true)
+        }
+    }
+}
```
Интересный способ)

Еще есть такой с фабричными методами:
```
struct CommentAppearance {
    static let standard = CommentAppearance(textColor: .gray, backgroundColor: .clear)
    static let red = CommentAppearance(textColor: .white, backgroundColor: .red)

    let textColor: UIColor
    let backgroundColor: UIColor
}
```

ios-review commented on DZ10/DZ10/Table/SettingTableViewCell.swift:
```diff
@@ -12,6 +12,11 @@ class SettingTableViewCell: UITableViewCell {
     
     @IBOutlet weak var titleLabel: UILabel!
     @IBOutlet weak var commentLabel: UILabel!
+        {
+        didSet {
+            
+        }
+    }
```
Зачем?

ios-review reviewed (approved):
Все классно👍
Молодец.

Можно мержить.
Реализация упрощений на ваше усмотрение.

angryQ committed:
Delete .DS_Store

angryQ committed:
Delete .DS_Store

angryQ committed:
Delete .DS_Store

angryQ committed:
test
